arelle/FileSource.py

```diff
@@ -655,6 +655,30 @@ def file(
         else:
             return openXmlFileStream(self.cntlr, filepath, stripDeclaration)
 
+    def getBytesSize(self) -> int | None:
+        """
+        Get the size of the zip file in bytes.
+        :return: Size of the zip file in bytes, or None if not applicable.
+        """
+        if isinstance(self.basefile, str) and os.path.isfile(self.basefile):
+            return os.path.getsize(self.basefile)
+        # ZipFile.fp is a private field, but is currently the simplest way for us to
+        # access the internal stream
+        if isinstance(self.fs, zipfile.ZipFile) and (fp := getattr(self.fs, 'fp')) is not None:
+            stream = cast(IO[Any], fp)
+            stream.seek(0, 2)  # Move to the end of the file
+            return stream.tell()  # Report the current position, which is the size of the file
+        return None
+
+    def getBytesSizeEstimate(self) -> int | None:
```

Comment:
This function name should probably indicate that it's for zip file sources.

Reply:
It doesn't seem like the function has to be specific to zip file sources (even if that's currently the only estimate).

Reply:
I don't think they do need to do that.  The intent is: call `getBytesSize`, if `None` then exact size is unknown, so call `getBytesSizeEstimate`, if `None` then not even an estimate is known.  Callers shouldn't have to know in what cases the exact or estimated sizes are available.

Reply:
That's fair